crates/node-core/src/metrics/prometheus_exporter.rs

```diff
@@ -67,10 +75,18 @@ async fn start_endpoint<F: Hook + 'static>(
             }))
         }
     });
+
     let server =
         Server::try_bind(&listen_addr).wrap_err("Could not bind to address")?.serve(make_svc);
 
-    tokio::spawn(async move { server.await.expect("Metrics endpoint crashed") });
+    task_executor.spawn_with_graceful_shutdown_signal(move |signal| async move {
+        server
+            .with_graceful_shutdown(async move {
+                let _ = signal.await;
+            })
+            .await
+            .expect("Metrics endpoint crashed")
```

Comment:
i'd rather print an error than panic here

Reply:
sure, this just mimicked the existing behavior. will change